common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -332,9 +344,44 @@ public MicrosoftStsTokenRequest createTokenRequest(@NonNull final MicrosoftStsAu
         tokenRequest.setClientId(request.getClientId());
         tokenRequest.setScope(request.getTokenScope());
         tokenRequest.setClaims(request.getClaims());
-
         tokenRequest.setGrantType(TokenRequest.GrantTypes.AUTHORIZATION_CODE);
+        setTokenRequestCorrelationId(tokenRequest);
+
+        if (SCHEME_POP.equals(authScheme.getName())) {
+            if (null == mStrategyParameters.getContext()) {
+                throw new ClientException(
+                        MicrosoftStsOAuth2Strategy.class.getSimpleName()
+                                + "Cannot execute PoP request sans Context"
+                );
+            }
+
+
+            // Add a token_type
+            tokenRequest.setTokenType(TokenRequest.TokenType.POP);
+
+            final IDevicePopManager devicePopManager = Device.getDevicePoPManagerInstance();
+
+            // Generate keys if they don't already exist...
+            if (!devicePopManager.asymmetricKeyExists()) {
+                final String thumbprint = devicePopManager.generateAsymmetricKey(mStrategyParameters.getContext());
+
+                Logger.verbosePII(
+                        TAG,
+                        "Generated new PoP asymmetric key with thumbprint: "
+                                + thumbprint
+                );
+            }
+
+            // Set the req_cnf
```

Comment:
This comment should be on the line below

or may be change this one to `// Get the req_cnf`

Reply:
resolved in fb60154e97bc7d16a768347af8650e968f1d6f11